src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs

```diff
@@ -25,4 +27,18 @@ public static ITextDecryptor CreateEncryptor(ConfigServerEncryptionSettings sett
 
         return decryptor;
     }
+
+    private static void EnsureValidEncryptionSettings(ConfigServerEncryptionSettings settings)
+    {
+        bool validEncryptionKeySettings = !string.IsNullOrEmpty(settings.EncryptionKey);
+
+        bool validKeystoreSettings = !string.IsNullOrEmpty(settings.EncryptionKeyStoreLocation) && !string.IsNullOrEmpty(settings.EncryptionKeyStorePassword) &&
+            !string.IsNullOrEmpty(settings.EncryptionKeyStoreAlias);
+
+        if (!(validEncryptionKeySettings ^ validKeystoreSettings))
```

Comment:
```suggestion
        if (validEncryptionKeySettings && validKeystoreSettings)
```
This reads slightly clearer to me , but feel free to leave as is if you feel strongly about it.

Reply:
Funny, I made a similar comment yesterday. Then I read the error message, realized the XOR was intentional, and deleted my comment. The code is correct and concise. Yet somehow it triggered both of us. Personally, I'd rewrite this to `&&`/`||` logical operators to improve code readability. Exclusive ORs (and to a lesser degree, bitwise operators in general) are more esoteric. I'm fine with keeping it this way, though.